androguard/core/apk/__init__.py

```diff
@@ -2032,7 +2201,7 @@ def get_certificates_v2(self) -> list[x509.Certificate]:
         """
         return [ x509.Certificate.load(cert) for cert in self.get_certificates_der_v2()]
 
-    def get_certificates_v1(self) -> list[x509.Certificate]:
+    def get_certificates_v1(self) -> list[Union[x509.Certificate, None]]:
```

Comment:
This will fail for `None` values. It should append those directly and not pass them to `x509.Certificate.load()`. Or maybe they should simply be skipped here?

```py
        for x in self.get_signature_names():
            certs.append(x509.Certificate.load(self.get_certificate_der(x)))
```

Reply:
I personally prefer `Optional[x]` to `Union[x, None]`. But up to you :)

Reply:
Also: `get_certificates()` below will fail for `None` values.